src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -3,88 +3,160 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections;
+using System.Collections.Generic;
 using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
 
 namespace System.Net
 {
     internal class HeaderInfoTable
     {
-        private static Hashtable HeaderHashTable;
-        private static HeaderInfo UnknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false);
+        private static readonly Func<string, string[]> s_singleParser = value => new[] { value };
+        private static readonly Func<string, string[]> s_multiParser = value => ParseValueHelper(value, isSetCookie: false);
+        private static readonly Func<string, string[]> s_setCookieParser = value => ParseValueHelper(value, isSetCookie: true);
+        private static readonly HeaderInfo s_unknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false, s_singleParser);
+        private static readonly Hashtable s_headerHashTable;
+        private static readonly Stack<char> s_valueParser = new Stack<char>();
+
+        private static string[] ParseValueHelper(string value, bool isSetCookie)
+        {
+            // RFC 6265: (for Set-Cookie header)
+            // If the name-value-pair string lacks a %x3D ("=") character, ignore the set-cookie-string entirely.
+            if (isSetCookie && (value.IndexOf('=') < 0)) return new string[0];
```

Comment:
`new string[0]` => `Array.Empty<string>()`

Reply:
Will fix.

Reply:
Because it will return a static object that is "shared" by everyone.

If you do a 'new' of anything it always allocates memory for a new object.